tfrecorder/schema.py

```diff
@@ -0,0 +1,179 @@
+# Lint as: python3
+
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Defines input types for tfrecorder's input schema. """
+
+import collections
+from typing import Dict, List, Union
+
+import frozendict
+import tensorflow as tf
+import tensorflow_transform as tft
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import schema_utils
+
+# All supported types will be based on _supported_type.
+_supported_type = collections.namedtuple(
```

Comment:
nit: Change this to CamelCase, e.g. SupportedType.
Reference: https://docs.python.org/3/library/collections.html#collections.namedtuple


possible future refactor: Consider using dataclasses or custom typehints

Reply:
Added a TODO for this.